### PR TITLE
Updated Backends.md to include compileWithoutConstants.

### LINKED ISSUE #2226: [Docs] Need to update the Backends section to reflect the new APIs
It looks like following section is not up to date with the recent changes done in compiledFunction class:

https://github.com/pytorch/glow/blob/master/docs/Backends.md#compiledfunction-abstract-class

### PR DESCRIPTION
**Pull request by gcatron**
*Description*: Added section about compileWithoutConstants, removed context from both compile function signatures.
*Testing*: Documentation only change.
*Documentation*: Updated  
Fixes #2226

### CODE REVIEW

File: docs/Backends.md [hunk 1]
```diff
@@ -20,15 +20,21 @@ All backends in Glow derive from the [abstract base class
...
-    should return a unique pointer to the
+  - This function takes a `Function *F` to compile. It should return a unique pointer to the
     [`CompiledFunction`](#compiledfunction-abstract-class) of `F`. If the backend
     uses Glow low-level IR, it can call `generateAndOptimizeIR()` to generate an
     optimized `IRFunction`.
 
+- `virtual std::unique_ptr<CompiledFunction> compileWithoutConstants(Function *F) const;`
+
+  - This function takes a `Function *F` to compile. This function does not collect
+    Constants this allows for   that action to be handled later. It should return
```
Comment (rdzhabarov) [inline docs/Backends.md]: too many spaces :)

Comment (rdzhabarov) [inline docs/Backends.md]: `collect Constants` is it described anywhere? What exactly that is.

Comment (gcatron) [reply docs/Backends.md]: I added a new section for RuntimeBundle and some more explanation. Should the Bundle go into a different document? It's sorta the bridge between Runtime and Compile time.

**Comment by gcatron** [comment]
@nickgg I'm pulling you in so this can resolve #2226. The compiledFunction section looked good to me but I wanted your opinion.

File: docs/Backends.md [hunk 2]
```diff
@@ -97,13 +101,25 @@ Additionally, there are virtual functions that backends can override:
...
+function returns, all Placeholders for the outputs of the function should have had 
+their backing tensor updated.
+
+### `RuntimeBundle` Helper Class
+
+`RuntimeBundle` is a helper class that contains the symbol information and collection
+of constants needed at runtime. This allows a function to be compiled without being linked
+to a context, and allows the `Function` to be freed after compilation. The symbol information
+is stored in a table where the key is the symbol name and the payload contains symbol information
+including, size, offset, input/output, and type. `RuntimeBundle` also contains a pointer to a
```
Comment (nickgg) [inline docs/Backends.md]: Not totally clear what `input/output` means here, maybe be more explicit.

File: docs/Backends.md [hunk 3]
```diff
@@ -97,13 +101,25 @@ Additionally, there are virtual functions that backends can override:
...
+block of memory that contains the constants for the `CompiledFunction`.
```
Comment (nickgg) [inline docs/Backends.md]: "that may contain?" it's not necessary for all backends to do this.

**Pull request merged**